probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -369,7 +368,7 @@ impl ArmProbeInterface for ArmCommunicationInterface<Initialized> {
     }
 
     fn current_debug_port(&self) -> DpAddress {
-        self.state.current_dp
+        self.state.current_dp.expect("A DpAddress is selected")
```

Comment:
```suggestion
        self.state.current_dp.expect("A DpAddress is selected. This is a bug, please report it.")
```

If this is an operational error in the probe-rs lib that would mean the logic is wrong, can we please add the hint about the bug? :) If this can actually be used wrong (I don't think so since it's private) by a library user we should avoid the expect.
Maybe we could add a hint for developers where this is initialized? :)